core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringParser.java

```diff
@@ -231,6 +234,8 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
         if (minimumShouldMatch != null && query instanceof BooleanQuery) {
             Queries.applyMinimumShouldMatch((BooleanQuery) query, minimumShouldMatch);
         }
+        
+        query.setBoost(boost);
```

Comment:
I think `query` can be null here in 1.x and before, we should double check before calling a method on it.